Lib/functools.py

```diff
@@ -984,6 +984,7 @@ def __init__(self, func):
         self.func = func
         self.attrname = None
         self.__doc__ = func.__doc__
+        self.__module__ = func.__module__
```

Comment:
Could you also add a unit test to `test_functools.py` asserting that `self.__module__` is set correctly on `cached_property` objects?

Reply:
Added! Thanks for all the suggestions @AlexWaygood 🙂